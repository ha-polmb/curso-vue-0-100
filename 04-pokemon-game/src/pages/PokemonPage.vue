<template>
  <div>
    <h1>¿Qué Pokémon es?</h1>

    <PokemonPicture :pokemonId="6" :showPokemon="true" />
    <PokemonOptions :pokemons="pokemonArr" />

  </div>
</template>

<script>
import PokemonOptions from '../components/PokemonOptions.vue'
import PokemonPicture from '../components/PokemonPicture.vue'
import getPokemonOptions from '../helpers/getPokemonOptions.js'

export default {
    components: {
        PokemonOptions,
        PokemonPicture
    },
    data() {
      return {
        pokemonArr: [],
      }
    },
    methods: {
      async mixPokemonArray() {
        this.pokemonArr = await getPokemonOptions()
      },
    },
    mounted() {
      this.mixPokemonArray()
    },
}
</script>