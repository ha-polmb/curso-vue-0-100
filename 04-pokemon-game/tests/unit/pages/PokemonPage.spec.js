import { shallowMount, mount } from "@vue/test-utils"
import PokemonPage from "@/pages/PokemonPage.vue"
import { pokemons } from "../mocks/pokemons.mock"


describe("PokemonPage component", () => {
    let wrapper

    beforeEach(() => {
        wrapper = shallowMount(PokemonPage)
    })

    test("debe llamar mixPokemonArray en mount", () => {
        const mixPokemonArraySpy = jest.spyOn(PokemonPage.methods, "mixPokemonArray")
        const wrapper = shallowMount(PokemonPage)
        expect(mixPokemonArraySpy).toHaveBeenCalled()
    })

    test("match con snapshot cuando cargan pokemons", () => {
        const wrapper = shallowMount(PokemonPage, {
            data() {
                return {
                    pokemonArr: pokemons,
                    pokemon: pokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ""
                }
            }
        })

        expect(wrapper.html()).toMatchSnapshot()
    })

    test("se muestran los componentes PokemonPicture y PokemonOptions", () => {
        const wrapper = shallowMount(PokemonPage, {
            data() {
                return {
                    pokemonArr: pokemons,
                    pokemon: pokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ""
                }
            }
        })
        const picture = wrapper.find("pokemon-picture-stub")
        const options = wrapper.find("pokemon-options-stub")
        expect(picture.exists()).toBeTruthy()
        expect(options.exists()).toBeTruthy()
        expect(picture.attributes("pokemonid")).toBe("1")
        expect(options.attributes("pokemons")).toBeTruthy()
    })

    test("pruebas con checkAnswer", async () => {
        const wrapper = shallowMount(PokemonPage, {
            data() {
                return {
                    pokemonArr: pokemons,
                    pokemon: pokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ""
                }
            }
        })

        await wrapper.vm.checkAnswer(1)
        expect(wrapper.find("h2").exists()).toBeTruthy()
        expect(wrapper.vm.showPokemon).toBeTruthy()
        expect(wrapper.find("h2").text()).toBe(`Correcto! ${pokemons[0].name}`)

        await wrapper.vm.checkAnswer(10)
        expect(wrapper.vm.message).toBe(`Oops, era ${pokemons[0].name}`)
    })
})