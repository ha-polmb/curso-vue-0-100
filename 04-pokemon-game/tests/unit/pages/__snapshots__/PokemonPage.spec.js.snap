// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage component match con snapshot cuando cargan pokemons 1`] = `
<div>
  <div>
    <h1>¿Qué Pokémon es?</h1>
    <pokemon-picture-stub pokemonid="1" showpokemon="false"></pokemon-picture-stub>
    <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub>
    <!--v-if-->
  </div>
</div>
`;
